### DISCUSSION TITLE
TypeScript: Information on NextComponentType and NextPage?

### DISCUSSION THREAD

**Discussion by danerwilliams** [vercel/next.js#39930] [Help]
I have had a really hard time finding authoritative information on the `NextComponentType` and `NextPage` types.

In the [TS section of the docs](https://nextjs.org/docs/basic-features/typescript), there seems to be no mention of either of these types. Searching for these types gives no results for `NextComponentType` and only gives [one mention](https://nextjs.org/docs/api-reference/data-fetching/get-initial-props#typescript) of `NextPage` related to `getInitialProps`. 

In most examples, they are never used. This has me wondering what is the point of them? What do they offer that `React.FC` type does not?

Finally, when is the appropriate use case for each? I think this can be inferred by the names, but it would nice to have it documented in writing somewhere.

**Comment by icyJoseph** [comment]
Typing an entire function is one acceptable path, but we could also leverage type inference built into the language. You type the arguments going into the component, and the type inferences calculates that the result is a JSX element or what not.

I for once, avoid using FC as well, but use it when I see fit, no rule is written in stone.

To your point, these two types give you access to the `getInitialProps` property. React components do not normally have this property on themselves. `NextComponentType` makes this property possible. For example, React components do have `displayName` on themselves.

More over, `NextPage` builds upon `NextComponentType`, by providing the type definition for the `getInitialProps` context argument. That means that `NextComponentType` is a blue print for a component that'll have a `getInitialProps` property, and keeps the arguments to this function generic. 

It is `NextPage` that makes the `getInitialProps` arguments specific.

**Reply by markomitranic** [reply]
I wonder how this fits into the app router. Seeing as the function names are now different, and that components can be async, it seems like these types are obsolete now. They also do not appear in App Router documentation

**Reply by hjdskhooba** [reply]
I totally agree with you @markomitranic
And I have the same problem with async components 😂

**Comment by Sargis686** [comment]
By using NextPage, you're explicitly indicating that this component represents a Next.js page. This type definition includes some Next.js-specific functionalities and can handle Next.js features like getInitialProps (used for server-side rendering) and other Next.js-specific properties. It also inherits the React.FC properties for handling props and rendering React components.

Overall, NextPage serves as a helpful typing for Next.js page components in TypeScript.
